scripts/ev_validation.py

```diff
@@ -0,0 +1,183 @@
+import dolfin as dlfn
+from stabilization import *
+import numpy as np
+import matplotlib.pyplot as plt
+
+dlfn.parameters['form_compiler']['no-evaluate_basis_derivatives'] = False
+
+def maximum(func, cell_i, dim):
+    V = func.function_space()
+
+    dmap = np.reshape(cell_i.get_coordinate_dofs(), (-1, dim))
+    coords = V.tabulate_dof_coordinates()
+
+    idx = np.fromiter((np.sum(np.product(coord==dmap, axis=1), axis=0) for coord in coords), dtype=bool)
+    return np.max(np.abs(func.vector().get_local()[idx]))
+
+def ev(J_n, J_n_, J_n_n, v, dt, h_, mesh_):
+    nu = .25 / dt * (3 * J_n**2 - 4 * J_n_**2 + J_n_n**2) + dlfn.dot(dlfn.grad(.5 * J_n**2), v)
+    return dlfn.project(h_ ** 2 * nu, dlfn.FunctionSpace(mesh_, 'CG', 2))
+
+#mesh = dlfn.UnitCubeMesh(5, 10, 20)
+x_0 = np.array([0., 0., 0.])
+x_1 = np.array([.0125, .0125, .0125])
+nx, ny, nz = 1, 1, 1
+
+meshes = [dlfn.IntervalMesh(nx, x_0[0], x_1[0]),
+        dlfn.RectangleMesh(dlfn.Point(x_0[:2]), dlfn.Point(x_1[:2]), nx, ny),
+        dlfn.BoxMesh(dlfn.Point(x_0), dlfn.Point(x_1), nx, ny, nz)]
+
+x_diff = x_1 - x_0
+h_min = [x_diff[0] / nx,
+         min(x_diff[0] / nx, x_diff[1] / ny),
+         min(x_diff[0] / nx, x_diff[1] / ny, x_diff[2] / nz)]
```

Comment:
This way of defining `h_min` is probably due to the way it is done in Guermond (2011). There is also method something like `h_min = mesh.hmin()`.